cmd/migrate.go

```diff
@@ -0,0 +1,287 @@
+					"mode":              mode,
+				},
+				Secrets: generationWorkflow.Jobs.Generate.Secrets,
+			},
+		},
+	}
+
+	/*
+	 * CREATE SOURCES
+	 */
+
+	docLocations := []string{}
+	if docs, ok := generationWorkflow.Jobs.Generate.With["openapi_docs"]; ok {
+		docLocations = append(docLocations, docs.([]string)...)
+	} else if docLocation, ok := generationWorkflow.Jobs.Generate.With["openapi_doc_location"]; ok {
+		docLocations = append(docLocations, docLocation.(string))
+	}
+
+	header := ""
+	if auth, ok := generationWorkflow.Jobs.Generate.With["openapi_doc_auth_header"]; ok {
+		header = auth.(string)
+	}
+	_, hasToken := generationWorkflow.Jobs.Generate.Secrets["openapi_doc_auth_token"]
+
+	sources := map[string]workflow.Source{
+		defaultSourceName: {
+			Inputs: docLocationsToDocuments(docLocations, header, hasToken),
+		},
+	}
+
+	/*
+	 * CREATE TARGETS
+	 */
+
+	languages := generationWorkflow.Jobs.Generate.With["languages"].(string)
+	langToOutput, err := parseLanguages(languages)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	targets := map[string]workflow.Target{}
+	for lang, output := range langToOutput {
+		targets[fmt.Sprintf("%s-target", lang)] = workflow.Target{
+			Source:     defaultSourceName,
+			Target:     lang,
+			Output:     output,
+			Publishing: getPublishing(generationWorkflow, lang),
+		}
+	}
+
+	workflowFile := &workflow.Workflow{
+		Version: "1.0.0",
+		Sources: sources,
+		Targets: targets,
+	}
+
+	return workflowFile, newGenWorkflow, nil
+}
+
+func parseLanguages(langString string) (map[string]*string, error) {
+	languageElems, err := getAllSubmatches(regexp.MustCompile(`\s*?- (.*)`), langString)
+	if err != nil {
+		return nil, err
+	}
+
+	languageToSubdirectory := map[string]*string{}
+	for _, languageElem := range languageElems {
+		// Languages are in one of the following formats:
+		// - go: ./path-to-go
+		// - go
+		if strings.Contains(languageElem, ":") {
+			languageAndSubdirectory := strings.Split(languageElem, ": ")
+			if len(languageAndSubdirectory) != 2 { //nolint:gomnd
+				return nil, fmt.Errorf("incorrectly formatted language in workflow file. Culprit: %s", languageElem)
+			}
+			languageToSubdirectory[languageAndSubdirectory[0]] = &languageAndSubdirectory[1]
+		} else {
+			languageToSubdirectory[languageElem] = nil
+		}
+	}
+
+	return languageToSubdirectory, nil
+}
+
+func docLocationsToDocuments(docLocations []string, authHeader string, hasToken bool) []workflow.Document {
+	var auth *workflow.Auth
+	if hasToken {
+		auth = &workflow.Auth{
+			Header: authHeader,
+			Secret: "$OPENAPI_DOC_AUTH_TOKEN", // This is the name of the secret in the action, which is fixed
+		}
+	}
+
+	var documents []workflow.Document
+	for _, docLocation := range docLocations {
+		documents = append(documents, workflow.Document{
+			Location: docLocation,
+			Auth:     auth,
+		})
+	}
+	return documents
+}
+
+func getPublishing(genWorkflow config.GenerateWorkflow, lang string) *workflow.Publishing {
```

Comment:
It doesn't look like this will cover the case where someone is in PR mode.

Reply:
Unless something has changed recently, you actually have to have `publish_*` in both workflow files